### PR TITLE
integrated with twelvedata api and implemented searchbyStock and sear…

### PR DESCRIPTION
**Pull request by Tokhirbek7**
…chbySymbo. Plus stocks are saved to the db on a scheduled tiem

### CODE REVIEW

File: src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                    connection.rollback();
+                }
+            } catch (SQLException rollbackException) {
+                rollbackException.printStackTrace();
+            }
+            e.printStackTrace();
+        } finally {
+            try {
+                if (connection != null) {
+                    connection.setAutoCommit(true);
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java:49]: for what we are doing this?

File: src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+public class StockDataRepository {
+    private final DataSource dataSource;
+    private final String SLQ_INSERT ="INSERT INTO Stock (symbol, name, currency, exchange, mic_code, country, type, date) " +
+            "VALUES (?, ?, ?, ?, ?, ?, ?, ?)";
+    public StockDataRepository(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
+    public void save(Stock stock) {
+        Connection connection = null;
+        try {
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java:21]: use try with resources

File: src/main/java/com/vention/stock_market_share/service/DailyStockUpdaterService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+            stock.setCountry(datum.getCountry());
+            stock.setType(datum.getType());
+            stock.setDate(new Date());
+            stockList.add(stock);
+        }
+        for (Stock value : stockList) {
+            stockDataRepository.save(value);
+        }
+    }
+
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/service/DailyStockUpdaterService.java:43]: no need this empty line

File: src/main/java/com/vention/stock_market_share/service/TwelveDataService.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+    public TwelveDataService(TwelveApiClient twelveApiClient) {
+        this.twelveApiClient = twelveApiClient;
+    }
+
+    public TwelveDataApiResponse searchStockBySymbol(String symbol) {
+        try {
+            return twelveApiClient.searchStockBySymbol(symbol);
+        } catch (TwelveDataApiException e) {
+            throw new TwelveDataApiException("Error while interacting with AlphaVantage API", e);
```
Comment (Poviraevsergei) [inline src/main/java/com/vention/stock_market_share/service/TwelveDataService.java:20]: we should not throw errors in the catch block, only logging.

**Review by Poviraevsergei — CHANGES_REQUESTED** [review]

File: src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java (hunk 1 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java:49]: I used transactions here. After reviewing my code i realised this part is not needed, I fixed it in my code

File: src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java (hunk 2 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/repository/StockDataRepository.java:21]: used

File: src/main/java/com/vention/stock_market_share/service/DailyStockUpdaterService.java (hunk 3 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/service/DailyStockUpdaterService.java:43]: empty line is deleted

File: src/main/java/com/vention/stock_market_share/service/TwelveDataService.java (hunk 4 above)
Comment (Tokhirbek7) [reply src/main/java/com/vention/stock_market_share/service/TwelveDataService.java:20]: logger.error() is used instead of throwing exception

**Pull request merged**